src/Generator/Hydrator/Property/SymfonyPropertyAccessorHydrator.php

```diff
@@ -55,6 +58,23 @@ public function hydrate(ObjectInterface $object, Property $property, GenerationC
     {
         $instance = $object->getInstance();
 
+        // check if property is an enum (might not have a type)
+        try {
+            $reflectionType = (new ReflectionProperty($instance, $property->getName()))->getType();
+
+            if (null !== $reflectionType && enum_exists($reflectionType->getName())) {
```

Comment:
I would rather do this:

- do this block as a fallback (i.e. only in case of failure); it is reflection heavy and we are doing it on a property basis, so if we can skip it it's best to do so. In case of failure, you can then throw the original error
- for readability I would split the block in two methods: `isEnumValue()` and `castValueToEnum()`

Reply:
👍 

Named it `isEnumProperty` since it made more sense (it checks the property not it's value)

Not 100% sure this is what you meant with those functions.